Deployment config is not set for system tests

### Description / Background

The system tests are failing because UploadDockerImages is being created without a DeployConfiguration.

This seems to have been caused by the following change:
- https://github.com/gchq/sleeper/pull/7114

### Screenshots/Logs

```
java.lang.NoClassDefFoundError: Could not initialize class sleeper.systemtest.suite.testutil.AwsSleeperSystemTestExtension
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:992)
	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:762)
	at java.base/java.util.stream.ReferencePipeline$7$1.accept(ReferencePipeline.java:276)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197)
	at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:150)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
Caused by: java.lang.ExceptionInInitializerError: Exception java.lang.NullPointerException: deployConfig must not be null [in thread "main"]
	at java.base/java.util.Objects.requireNonNull(Objects.java:235)
	at sleeper.clients.deploy.container.UploadDockerImages.<init>(UploadDockerImages.java:57)
	at sleeper.clients.deploy.container.UploadDockerImages$Builder.build(UploadDockerImages.java:206)
	at sleeper.systemtest.drivers.instance.AwsSleeperInstanceDriver.<init>(AwsSleeperInstanceDriver.java:79)
	at sleeper.systemtest.drivers.util.AwsSystemTestDrivers.instance(AwsSystemTestDrivers.java:110)
	at sleeper.systemtest.dsl.instance.SystemTestDeploymentContext.<init>(SystemTestDeploymentContext.java:32)
	at sleeper.systemtest.suite.testutil.AwsSleeperSystemTestExtension.<clinit>(AwsSleeperSystemTestExtension.java:27)
	... 15 more
```

--- a/java/system-test/system-test-drivers/src/main/java/sleeper/systemtest/drivers/instance/AwsSleeperInstanceDriver.java
+++ b/java/system-test/system-test-drivers/src/main/java/sleeper/systemtest/drivers/instance/AwsSleeperInstanceDriver.java
@@ -27,11 +27,6 @@ import software.amazon.awssdk.services.s3.S3Client;
 import sleeper.clients.deploy.DeployExistingInstance;
 import sleeper.clients.deploy.DeployInstance;
 import sleeper.clients.deploy.DeployNewInstance;
-import sleeper.clients.deploy.container.UploadDockerImages;
-import sleeper.clients.deploy.container.UploadDockerImages.CopyContainerImage;
-import sleeper.clients.deploy.container.UploadDockerImagesToEcr;
-import sleeper.clients.deploy.jar.SyncJars;
-import sleeper.clients.util.cdk.InvokeCdk;
 import sleeper.configuration.properties.S3InstanceProperties;
 import sleeper.core.deploy.SleeperInstanceConfiguration;
 import sleeper.core.properties.instance.InstanceProperties;
@@ -68,21 +63,7 @@ public class AwsSleeperInstanceDriver implements SleeperInstanceDriver {
         this.dynamoDB = clients.getDynamo();
         this.cloudFormationClient = clients.getCloudFormation();
         this.resetInstance = new AwsResetInstanceOnFirstConnect(clients);
-        this.deployInstance = new DeployInstance(
-                SyncJars.fromScriptsDirectory(s3, parameters.getAccount(), parameters.getScriptsDirectory()),
-                new UploadDockerImagesToEcr(
-                        UploadDockerImages.builder()
-                                .scriptsDirectory(parameters.getScriptsDirectory())
-                                .copyImage(CopyContainerImage.withTransferManager(clients.getEcr()))
-                                .commandRunner(clients.getCommandRunner())
-                                .createMultiplatformBuilder(parameters.isCreateMultiPlatformBuilder())
-                                .build(),
-                        parameters.getAccount(), clients.getRegionProvider().getRegion().id()),
-                DeployInstance.WriteLocalProperties.underScriptsDirectory(parameters.getScriptsDirectory()),
-                InvokeCdk.builder()
-                        .scriptsDirectory(parameters.getScriptsDirectory())
-                        .runCommand(clients.getCommandRunner())
-                        .build());
+        this.deployInstance = SystemTestDeploymentFactory.createDeployInstance(parameters, clients);
     }
 
     public void loadInstanceProperties(InstanceProperties instanceProperties, String instanceId) {

--- a/java/system-test/system-test-drivers/src/main/java/sleeper/systemtest/drivers/instance/AwsSystemTestDeploymentDriver.java
+++ b/java/system-test/system-test-drivers/src/main/java/sleeper/systemtest/drivers/instance/AwsSystemTestDeploymentDriver.java
@@ -20,19 +20,14 @@ import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 import software.amazon.awssdk.services.cloudformation.CloudFormationClient;
 import software.amazon.awssdk.services.cloudformation.model.CloudFormationException;
-import software.amazon.awssdk.services.ecr.EcrClient;
 import software.amazon.awssdk.services.s3.S3Client;
 
-import sleeper.clients.deploy.DeployConfiguration;
-import sleeper.clients.deploy.container.UploadDockerImages;
-import sleeper.clients.deploy.container.UploadDockerImages.CopyContainerImage;
 import sleeper.clients.deploy.container.UploadDockerImagesToEcr;
 import sleeper.clients.deploy.container.UploadDockerImagesToEcrRequest;
 import sleeper.clients.deploy.jar.SyncJars;
 import sleeper.clients.deploy.jar.SyncJarsRequest;
 import sleeper.clients.util.cdk.CdkCommand;
 import sleeper.clients.util.cdk.InvokeCdk;
-import sleeper.clients.util.command.CommandUtils;
 import sleeper.core.deploy.LambdaJar;
 import sleeper.core.properties.model.SleeperArtefactsLocation;
 import sleeper.systemtest.configuration.SystemTestStandaloneProperties;
@@ -58,13 +53,17 @@ public class AwsSystemTestDeploymentDriver implements SystemTestDeploymentDriver
     private final SystemTestParameters parameters;
     private final S3Client s3;
     private final CloudFormationClient cloudFormation;
-    private final EcrClient ecr;
+    private final SyncJars syncJars;
+    private final UploadDockerImagesToEcr dockerUploader;
+    private final InvokeCdk cdk;
 
     public AwsSystemTestDeploymentDriver(SystemTestParameters parameters, SystemTestClients clients) {
         this.parameters = parameters;
         this.s3 = clients.getS3();
         this.cloudFormation = clients.getCloudFormation();
-        this.ecr = clients.getEcr();
+        this.syncJars = SystemTestDeploymentFactory.createSyncJars(parameters, clients);
+        this.dockerUploader = SystemTestDeploymentFactory.createDockerUploader(parameters, clients);
+        this.cdk = SystemTestDeploymentFactory.createInvokeCdk(parameters, clients);
     }
 
     @Override
@@ -97,11 +96,8 @@ public class AwsSystemTestDeploymentDriver implements SystemTestDeploymentDriver
             Path generatedDirectory = Files.createDirectories(parameters.getGeneratedDirectory());
             Path propertiesFile = generatedDirectory.resolve("system-test.properties");
             deployProperties.save(propertiesFile);
-            InvokeCdk.builder()
-                    .jarsDirectory(parameters.getJarsDirectory())
-                    .runCommand(CommandUtils::runCommandLogOutput)
-                    .build().invoke(SYSTEM_TEST_INFRA,
-                            CdkCommand.deploySystemTestStandalone(propertiesFile));
+            cdk.invoke(SYSTEM_TEST_INFRA,
+                    CdkCommand.deploySystemTestStandalone(propertiesFile));
         } catch (InterruptedException e) {
             Thread.currentThread().interrupt();
             throw new RuntimeException(e);
@@ -111,27 +107,15 @@ public class AwsSystemTestDeploymentDriver implements SystemTestDeploymentDriver
     }
 
     private void uploadJarsAndDockerImages() throws IOException, InterruptedException {
-        InvokeCdk.builder()
-                .jarsDirectory(parameters.getJarsDirectory())
-                .runCommand(CommandUtils::runCommandLogOutput)
-                .build().invoke(ARTEFACTS,
-                        CdkCommand.deployArtefacts(parameters.getArtefactsDeploymentId(), List.of(SYSTEM_TEST_IMAGE.getImageName())));
-        new SyncJars(s3, parameters.getAccount(), parameters.getJarsDirectory())
-                .sync(SyncJarsRequest.builder()
-                        .deploymentId(parameters.getArtefactsDeploymentId())
-                        .uploadFilter(jar -> LambdaJar.isFileJar(jar, CUSTOM_RESOURCES))
-                        .build());
+        cdk.invoke(ARTEFACTS,
+                CdkCommand.deployArtefacts(parameters.getArtefactsDeploymentId(), List.of(SYSTEM_TEST_IMAGE.getImageName())));
+        syncJars.sync(SyncJarsRequest.builder()
+                .deploymentId(parameters.getArtefactsDeploymentId())
+                .uploadFilter(jar -> LambdaJar.isFileJar(jar, CUSTOM_RESOURCES))
+                .build());
         if (!parameters.isSystemTestClusterEnabled()) {
             return;
         }
-        UploadDockerImagesToEcr dockerUploader = new UploadDockerImagesToEcr(
-                UploadDockerImages.builder()
-                        .scriptsDirectory(parameters.getScriptsDirectory())
-                        .deployConfig(DeployConfiguration.fromScriptsDirectory(parameters.getScriptsDirectory()))
-                        .commandRunner(CommandUtils::runCommandLogOutput)
-                        .copyImage(CopyContainerImage.withTransferManager(ecr))
-                        .build(),
-                parameters.getAccount(), parameters.getRegion());
         dockerUploader.upload(UploadDockerImagesToEcrRequest.builder()
                 .ecrPrefix(SleeperArtefactsLocation.getDefaultEcrRepositoryPrefix(parameters.getArtefactsDeploymentId()))
                 .images(List.of(SYSTEM_TEST_IMAGE))

--- a/java/system-test/system-test-drivers/src/main/java/sleeper/systemtest/drivers/instance/SystemTestDeploymentFactory.java
+++ b/java/system-test/system-test-drivers/src/main/java/sleeper/systemtest/drivers/instance/SystemTestDeploymentFactory.java
@@ -1,0 +1,70 @@
+/*
+ * Copyright 2022-2026 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.systemtest.drivers.instance;
+
+import sleeper.clients.deploy.DeployConfiguration;
+import sleeper.clients.deploy.DeployInstance;
+import sleeper.clients.deploy.container.UploadDockerImages;
+import sleeper.clients.deploy.container.UploadDockerImages.CopyContainerImage;
+import sleeper.clients.deploy.container.UploadDockerImagesToEcr;
+import sleeper.clients.deploy.jar.SyncJars;
+import sleeper.clients.util.cdk.InvokeCdk;
+import sleeper.systemtest.drivers.util.SystemTestClients;
+import sleeper.systemtest.dsl.instance.SystemTestParameters;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+
+public class SystemTestDeploymentFactory {
+
+    private SystemTestDeploymentFactory() {
+    }
+
+    public static DeployInstance createDeployInstance(SystemTestParameters parameters, SystemTestClients clients) {
+        return new DeployInstance(
+                createSyncJars(parameters, clients),
+                createDockerUploader(parameters, clients),
+                DeployInstance.WriteLocalProperties.underScriptsDirectory(parameters.getScriptsDirectory()),
+                createInvokeCdk(parameters, clients));
+    }
+
+    public static SyncJars createSyncJars(SystemTestParameters parameters, SystemTestClients clients) {
+        return SyncJars.fromScriptsDirectory(clients.getS3(), parameters.getAccount(), parameters.getScriptsDirectory());
+    }
+
+    public static UploadDockerImagesToEcr createDockerUploader(SystemTestParameters parameters, SystemTestClients clients) {
+        try {
+            return new UploadDockerImagesToEcr(
+                    UploadDockerImages.builder()
+                            .scriptsDirectory(parameters.getScriptsDirectory())
+                            .deployConfig(DeployConfiguration.fromScriptsDirectory(parameters.getScriptsDirectory()))
+                            .commandRunner(clients.getCommandRunner())
+                            .copyImage(CopyContainerImage.withTransferManager(clients.getEcr()))
+                            .build(),
+                    parameters.getAccount(), parameters.getRegion());
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    public static InvokeCdk createInvokeCdk(SystemTestParameters parameters, SystemTestClients clients) {
+        return InvokeCdk.builder()
+                .scriptsDirectory(parameters.getScriptsDirectory())
+                .runCommand(clients.getCommandRunner())
+                .build();
+    }
+
+}

--- a/java/system-test/system-test-dsl/src/main/java/sleeper/systemtest/dsl/instance/SystemTestParameters.java
+++ b/java/system-test/system-test-dsl/src/main/java/sleeper/systemtest/dsl/instance/SystemTestParameters.java
@@ -122,14 +122,6 @@ public class SystemTestParameters {
         return scriptsDirectory;
     }
 
-    public Path getJarsDirectory() {
-        return scriptsDirectory.resolve("jars");
-    }
-
-    public Path getDockerDirectory() {
-        return scriptsDirectory.resolve("docker");
-    }
-
     public Path getGeneratedDirectory() {
         return scriptsDirectory.resolve("generated");
     }

